wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -730,44 +741,160 @@ public void init(InitRequest request) throws GitException {
             while (revIterator.hasNext()) {
                 RevCommit commit = revIterator.next();
-                PersonIdent committerIdentity = commit.getCommitterIdent();
-
-                GitUser gitUser = newDto(GitUser.class).withName(committerIdentity.getName())
-                                                       .withEmail(committerIdentity.getEmailAddress());
-
-                Revision revision = newDto(Revision.class).withId(commit.getId().getName())
-                                                          .withMessage(commit.getFullMessage())
-                                                          .withCommitTime(MILLISECONDS.convert(commit.getCommitTime(), SECONDS))
-                                                          .withCommitter(gitUser);
+                Revision revision = getRevision(commit, filePath);
                 commits.add(revision);
             }
             return new LogPage(commits);
         } catch (GitAPIException | IOException exception) {
-            String errorMessage = exception.getMessage();
-            if (ERROR_LOG_NO_HEAD_EXISTS.equals(errorMessage)) {
-                throw new GitException(errorMessage, ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED);
+			String errorMessage = exception.getMessage();
+			LOG.error("Failed to retrieve log. ", errorMessage);
+			if (ERROR_LOG_NO_HEAD_EXISTS.equals(errorMessage)) {
+				throw new GitException(errorMessage, ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED);
+			} else {
+				throw new GitException(exception);
+			}
+        }
+    }
+
+    private Revision getRevision(RevCommit commit, String filePath) throws GitAPIException, IOException {
+        List<String> commitParentsList = Stream.of(commit.getParents())
+                                               .map(RevCommit::getName)
+                                               .collect(Collectors.toList());
+        List<Branch> commitBranchesList = getBranchesOfCommit(commit, ListMode.ALL);
+        List<DiffCommitFile> diffCommitFilesList = getCommitDiffFiles(commit, filePath);
+
+        GitUser author = getCommitAuthor(commit);
+        GitUser committer = getCommitCommitter(commit);
+
+        return newDto(Revision.class).withId(commit.getId()
+                                                     .getName())
+                                                     .withMessage(commit.getFullMessage())
+                                                     .withCommitTime((long) commit.getCommitTime() * 1000)
+                                                     .withCommitter(committer)
+                                                     .withAuthor(author)
+                                                     .withBranches(commitBranchesList)
+                                                     .withCommitParent(commitParentsList)
+                                                     .withDiffCommitFile(diffCommitFilesList);
```

Comment:
formatting (https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#for-dot--separator-break-always-comes-before-the-symbol)